lib/Onyx.js

```diff
@@ -804,7 +843,11 @@ function notifyCollectionSubscribersOnNextTick(key, value) {
 function remove(key) {
     cache.drop(key);
     notifySubscribersOnNextTick(key, null);
-    return Storage.removeItem(key);
+
+    const nextModification = () => Storage.removeItem(key)
+        .catch(error => Logger.logAlert(`An error occurred while removing key: ${key}, Error: ${error}`));
+
+    return waitPendingWrites(nextModification, key);
```

Comment:
Should we notifySubscribersOnNextTick after waiting for the pending writes instead of before? What happens if one of the pending writes sets the key? With the current changes would we notify that it was null, then non null, then actually set it to null?

Reply:
i think we generally want to notify before, because we're also setting the cache in `notify`. But in this case i agree that we probably shouldn't notify 3x times..

Reply:
With the current implementation, we're also notifying before actually setting the value, right?

Like [here](https://github.com/Expensify/react-native-onyx/blob/5a97bf5282e614b9e9357fb8d620815f0399cef5/lib/Onyx.js#LL871C45-L871C45)

Reply:
> With the current implementation, we're also notifying before actually setting the value, right?

Yes that's true. My concern is that previously we notify and then remove it from storage right away, now we remove it from storage after the pending writes have occurred.

But I suppose that's in line with how Onyx.set works in this implementation. I'm not sure about this honestly. I'm curious to hear what others think.